src/SMTLIB/Backends/Process.hs

```diff
@@ -48,9 +50,15 @@ data Config = Config
   { -- | The command to call to run the solver.
     exe :: String,
     -- | Arguments to pass to the solver's command.
-    args :: [String]
+    args :: [String],
+    -- | A function for logging the solver process' creation, errors and termination.
+    logger :: LBS.ByteString -> IO ()
   }
 
+instance Default Config where
+  -- By default, use Z3 as an external process and ignore log messages.
+  def = Config "z3" ["-in"] $ const $ return ()
```

Comment:
Yes, I figured it doesn't hurt to have a default solver so that the user trying out the library doesn't even have to find out how to use solvers from the command line.

> If this is to be shown in the generated haddock documentation it needs to be mentioned elsewhere.

Yeah the missing `|` is a typo. But  I should be doing
```haskell
  -- | By default, use Z3 as an external process and ignore log messages.
instance Default Config where
  def = Config "z3" ["-in"] $ const $ return ()
```
instead, right ?